pkg/storage/bloom/v1/ast_extractor.go

```diff
@@ -1,12 +1,22 @@
 package v1
 
 import (
+	regexsyn "github.com/grafana/regexp/syntax"
+
 	"github.com/prometheus/prometheus/model/labels"
 
 	"github.com/grafana/loki/v3/pkg/logql/log"
 	"github.com/grafana/loki/v3/pkg/logql/syntax"
+	"github.com/grafana/loki/v3/pkg/util"
 )
 
+// Simplifiable regexp expressions can quickly expand into very high
+// cardinality; we limit the number of matchers to prevent this.
+//
+// For example, the regex `[0-9]` expands to 10 matchers (0, 1, .. 9), while
+// `[0-9][0-9]` expands to 100 matchers (00, 01, .., 99).
+const maxRegexMatchers = 25
```

Comment:
Testing blooms is relatively cheap compared to all the burden of downloading blocks, and passing stuff around the network. I think we can affort a higher max here. What about 200?

Reply:
🤷 If both you and @chaudum think 200 is safe, I'm fine with it.

Reply:
Let's try it with 200, we can always reduce it